File: smartpizza-frontend/src/services/productService.js
import api from './api'

export const productService = {
  getAll: () => api.get('/products').then(r => r.data),
  getById: (id) => api.get(`/products/${id}`).then(r => r.data),
  getByCategory: (category) => api.get(`/products/category/${category}`).then(r => r.data),
  getBySize: (size) => api.get(`/products/size/${size}`).then(r => r.data),
  search: (keyword) => api.get(`/products/search/${keyword}`).then(r => r.data),
  getByPriceRange: (min, max) => api.get('/products/price-range', { params: { min, max } }).then(r => r.data),
  getTopExpensive: () => api.get('/products/top-expensive').then(r => r.data),
  getRandom: () => api.get('/products/random').then(r => r.data),
  add: (data) => api.post('/products', data).then(r => r.data),
  update: (id, data) => api.put(`/products/${id}`, data).then(r => r.data),
  delete: (id) => api.delete(`/products/${id}`).then(r => r.data)
}

File: SmartPizza-backend/src/main/java/com/wipro/ecom/services/ProductService.java
package com.wipro.ecom.services;

import java.util.List;

import org.springframework.data.jpa.repository.Query;

import com.wipro.ecom.dtos.ProductDTO;

public interface ProductService {

    ProductDTO addProduct(ProductDTO dto);

    List<ProductDTO> getAllProducts();

    ProductDTO getProductById(Long id);

    List<ProductDTO> getByCategory(String category);

    List<ProductDTO> getByPriceLessThan(double price);

    List<ProductDTO> getBySize(String size);

    List<ProductDTO> searchByName(String keyword);

    List<ProductDTO> getPriceRange(double min, double max);

    List<ProductDTO> getTopExpensive();

    List<ProductDTO> getTopCheap();
    
    List<ProductDTO> getAIRecommendations(String category);

    List<ProductDTO> getRandomProducts();

	ProductDTO updateProduct(Long id, ProductDTO dto);

	void deleteProduct(Long id);
}
